Dockerfile

```
@@ -87,11 +87,10 @@ RUN SPEAD2_VERSION=$(grep ^spead2== katgpucbf/requirements.txt | cut -d= -f3) &&
     wget "https://github.com/ska-sa/spead2/releases/download/v$SPEAD2_VERSION/spead2-$SPEAD2_VERSION.tar.gz" && \
     tar -zxf "spead2-$SPEAD2_VERSION.tar.gz" && \
     cd "spead2-$SPEAD2_VERSION" && \
-    mkdir build && \
+    meson setup build && \
```

I suggest changing this to
```suggestion
    meson setup -Dtools=disabled -Dibv=enabled -Dmlx5dv=enabled build && \
```
That will
- force the build to fail if ibv or mlx5dv is not found (in which case the rest of the Dockerfile hasn't installed things properly)
- skip building the command-line tools, which aren't needed for fsim

Then you should be able to remove boost-program-options-dev.